ansible/playbooks/do_teardown.yml

```diff
@@ -0,0 +1,160 @@
+---
+- name: teardown browsertrix cloud on digital ocean
+  hosts: localhost
+  connection: local
+  gather_facts: false
+  vars_files:
+    - ../group_vars/do/main.yml
+    - ../group_vars/do/teardown.yml
+
+  tasks:
+
+  # Init
+  # ===========================================
+    - name: d_ocean | init | set full domain
+      ansible.builtin.set_fact:
+        full_domain: "{{ subdomain + '.' + domain if subdomain else domain }}"
+
+    - name: d_ocean | init | install s3cmd and mongodump
+      ansible.builtin.package:
+        name: "{{ item }}"
+        state: present
+      loop:
+        - mongodump
+        - s3cmd
+
+  # MongoDB
+  # ===========================================
+    - name: d_ocean | db | test for existing mongodb
+      ansible.builtin.command: doctl db list -o json
+      changed_when: false
+      failed_when: false
+      register: db_check
+
+    - name: d_ocean | db | does db exist
+      ansible.builtin.set_fact:
+        db_exists: "{{ db_check.stdout | from_json | json_query(name_query) | length >= 1 }}"
+      vars:
+        name_query: '[?name==`{{ db_name }}`]'
+
+    - name: d_ocean | db | set db id
+      ansible.builtin.set_fact:
+        db_uuid: "{{ db_check.stdout | from_json | json_query(name_query) | json_query('[0].id') }}"
+      vars:
+        name_query: '[?name==`{{ db_name }}`]'
+
+    - name: d_ocean | db | reset db user password
+      ansible.builtin.command: doctl databases user reset {{ db_uuid }} doadmin -o json
+      register: db_user
+      when: db_exists
+      changed_when: true
+
+    - name: Sleep 10 seconds to wait for admin password to change
+      ansible.builtin.wait_for:
+        timeout: 10
+      when: db_exists
+
+    - name: d_ocean | db | set db config
+      ansible.builtin.set_fact:
+        db_url: "{{ db_check.stdout | from_json | json_query(name_query) | json_query('[0].connection.uri') | replace(old, new) }}"
+      vars:
+        name_query: '[?name==`{{ db_name }}`]'
+        old: ":@"
+        new: ":{{ db_user.stdout | from_json | json_query('[0].password') }}@"
+      when: db_exists
+
+    - name: d_ocean | db | dump db
+      ansible.builtin.command: mongodump --uri={{ db_url }} --archive=mongobackup.gz --gzip
+      when: create_backup and db_exists
+      changed_when: true
+
+    - name: d_ocean | db | upload mongo to spaces
+      ansible.builtin.command: s3cmd put ./mongobackup.gz s3://{{ db_name }} --host={{ droplet_region }}.digitaloceanspaces.com --access_key "{{ lookup('env', 'DO_AWS_ACCESS_KEY') }}" --secret_key "{{ lookup('env', 'DO_AWS_SECRET_KEY') }}"
```

Comment:
Instead of `s3://{{ db_name }}`, should this be the root of the path we're configured for storage,
which might `s3://{{ bucket_name }}/{{ bucket_path }}` ?

Reply:
Good catch! I think it's supposed to be `s3://{{ db_name }}/{{ bucket_path }}`? I'll check and make sure it's correct.

Reply:
I think we allow overriding the bucket also with custom `bucket_name`